### PR TITLE
chore: add batch size for index

### PR DESCRIPTION
**Pull request by gabrielfs7**
Control amount of documents can be send in the same request

https://oat-sa.atlassian.net/browse/ADF-256

- [ ] https://github.com/oat-sa/extension-tao-advanced-search/pull/19

### CODE REVIEW

File: scripts/tools/index/IndexPopulator.php [hunk 1]
```diff
@@ -189,5 +197,41 @@ protected function getScriptReport(int $result, string $class, int $limit, int $
...
+            if (count($paginatedResources[$batchIndex]) >= $batchSize) {
+                $batchIndex++;
+            }
+
+            $paginatedResources[$batchIndex][] = $resource;
+        }
+
+        foreach ($paginatedResources as $key => $resources) {
+            $indexIterator = new IndexIterator(new ResultSet($resources, count($resources)));
+            $indexIterator->setServiceLocator($this->getServiceLocator());
```
Comment (boajer) [inline scripts/tools/index/IndexPopulator.php:219]:
```suggestion
            $this->propagate($indexIterator);
```

File: scripts/tools/index/IndexPopulator.php [hunk 2]
```diff
@@ -58,6 +60,13 @@ protected function provideOptions(): array
                 'description' => 'The limit of resources to be processed per class',
                 'defaultValue' => 10
             ],
+            'indexBatchSize' => [
+                'prefix' => 'ibs',
+                'longPrefix' => 'limit',
+                'flag' => false,
+                'description' => 'The limit of resources to be processed per class',
```
Comment (boajer) [inline scripts/tools/index/IndexPopulator.php:67]:
```suggestion
                'description' => 'The batch size  of resources to be processed per class',
```

File: scripts/tools/index/IndexPopulator.php [hunk 3]
```diff
@@ -58,6 +60,13 @@ protected function provideOptions(): array
                 'description' => 'The limit of resources to be processed per class',
                 'defaultValue' => 10
             ],
+            'indexBatchSize' => [
+                'prefix' => 'ibs',
+                'longPrefix' => 'limit',
```
Comment (boajer) [inline scripts/tools/index/IndexPopulator.php:65]:
```suggestion
                'longPrefix' => 'classBatchSize',
```

File: scripts/tools/index/IndexPopulator.php (hunk 1 above)
Comment (gabrielfs7) [reply scripts/tools/index/IndexPopulator.php]: Done, thanks!

File: scripts/tools/index/IndexPopulator.php (hunk 2 above)
Comment (gabrielfs7) [reply scripts/tools/index/IndexPopulator.php]: Done, thanks!

File: scripts/tools/index/IndexPopulator.php (hunk 3 above)
Comment (gabrielfs7) [reply scripts/tools/index/IndexPopulator.php]: Done, thanks!

**Pull request closed without merging**